File: dfs/1457.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func pseudoPalindromicPaths (root *TreeNode) int {
    if root == nil {
        return 0
    }
    l := make([]int, 9)
    var ret int
    dfs(root, l, &ret)
    return ret
}

func dfs(root *TreeNode, l []int, ret *int) {
    l[root.Val-1]++
    if root.Left == nil && root.Right == nil {
        if checkPalin(l) {
            (*ret)++   
        }
    }else { 
        if root.Left != nil {
            dfs(root.Left, l, ret)
            l[root.Left.Val-1]--
        }
        if root.Right != nil {
            dfs(root.Right, l, ret)
            l[root.Right.Val-1]--
        }
    }
}

func checkPalin(l []int) bool {
    var count_odd, count_even int
    for _, i := range l {
        if i % 2 != 0 {
            count_odd++
        }else {
            count_even++
        }
    }
    if count_odd == 0 || count_odd == 1 && count_even > 0 {
        return true
    }
    return false
}

File: tree/1457.cc
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int pseudoPalindromicPaths (TreeNode* root) {
        std::vector<int> cur;
        int ret = 0;
        recur (root, cur, ret);
        return ret;
    }
private:
    void recur(TreeNode* root, std::vector<int> cur, int& ret) {
        if (!root->left && !root->right) {
            cur.emplace_back(root->val);
            if (check(cur))
                ++ret;
            return;
        }
        cur.emplace_back(root->val);
        if (root->left)
            recur(root->left, cur, ret);
        if (root->right)
            recur(root->right, cur, ret);
    }
    bool check(std::vector<int> cur) {

        std::unordered_map<int, int> m;
        for (const auto& i : cur) {
            auto iter = m.find(i);
            if (iter != m.end()) {
                ++iter->second;
            } else {
                m.emplace(i, 1);
            }
        }
       
        int count = 0;
        for (auto iter = m.begin(); iter != m.end(); ++iter) {
            if (iter->second % 2 != 0)
                ++count;
            if (count > 1) {
                return false;
            }
        }
        return true;
    }
};
